manager/dispatcher/dispatcher.go

```diff
@@ -1091,13 +1114,33 @@ func (d *Dispatcher) Assignments(r *api.AssignmentsRequest, stream api.Dispatche
 	}
 }
 
-func (d *Dispatcher) nodeRemove(id string, status api.NodeStatus) error {
+// markNodeNotReady sets the node state to some state other than READY
+func (d *Dispatcher) markNodeNotReady(id string, state api.NodeStatus_State, message string) error {
 	if err := d.isRunningLocked(); err != nil {
 		return err
 	}
 
+	// lookup the node so we can get its old address
+	// TODO(dperny) is there a faster/better way to do this?
```

Comment:
yeah i was thinking about this change in the shower this morning, this is definitely going to be the best way to do this.